### PR TITLE
Add dftx ICXSubmitDFCHTLC and TxnBuilderICXOrderBook.

### PR DESCRIPTION
**Pull request by surangap**
#### What kind of PR is this?:

/kind feature

#### What this PR does / why we need it:
Adds support to create ICXSubmitDFCHTLC custom transactions from jellyfish side.
#### Which issue(s) does this PR fixes?:

Fixes part of #533

### CODE REVIEW

File: packages/jellyfish-transaction-builder/__tests__/txn/txn_builder_submit_dfc_htlc.test.ts [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+    expect(outs[1].scriptPubKey.reqSigs).toStrictEqual(1)
+    expect(outs[1].scriptPubKey.addresses[0]).toStrictEqual(await providers.getAddress())
+
+    await container.generate(1)
+
+    // List htlc and check
+    const listHTLCOptions: ICXListHTLCOptions = {
+      offerTx: makeOfferTxId
+    }
+    const HTLCs: Record<string, ICXDFCHTLCInfo | ICXEXTHTLCInfo | ICXClaimDFCHTLCInfo> = await client.call('icx_listhtlcs', [listHTLCOptions], 'bignumber')
```
Comment (Jouzo) [inline packages/jellyfish-transaction-builder/__tests__/txn/txn_builder_submit_dfc_htlc.test.ts:88]: Consider refactor to
export type ListHTLCResult = Record<string, ICXDFCHTLCInfo | ICXEXTHTLCInfo | ICXClaimDFCHTLCInfo>

**Comment by Jouzo** [comment]
Looks good to me

File: packages/jellyfish-transaction-builder/__tests__/txn/txn_builder_submit_dfc_htlc.test.ts (hunk 1 above)
Comment (surangap) [reply packages/jellyfish-transaction-builder/__tests__/txn/txn_builder_submit_dfc_htlc.test.ts:88]: did not get it. why do you want to export it here?

**Review by canonbrother — COMMENTED** [review]
i'm good as well

**Review by jingyi2811 — DISMISSED** [review]

**Review by Jouzo — DISMISSED** [review]

File: packages/jellyfish-api-core/__tests__/category/icxorderbook/icx_setup.ts [hunk 2]
```diff
@@ -44,6 +44,11 @@ export class ICXSetup {
     accountBTC = await this.container.call('getnewaddress')
   }
 
+  async setAccounts (address1: string, address2: string): Promise<void> {
+    accountDFI = address1
+    accountBTC = address2
+  }
+
```
Comment (surangap) [inline packages/jellyfish-api-core/__tests__/category/icxorderbook/icx_setup.ts:51]: Redundant code.
```suggestion

```

File: packages/jellyfish-transaction-builder/__tests__/txn/txn_builder_submit_dfc_htlc.test.ts [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+import { icxorderbook } from '@defichain/jellyfish-api-core'
+
+describe('submit DFC HTLC', () => {
+  const container = new MasterNodeRegTestContainer()
+  const testing = Testing.create(container)
+  let providers: MockProviders
+  let builder: P2WPKHTransactionBuilder
+
+  beforeAll(async () => {
+    await container.start()
```
Comment (Jouzo) [inline packages/jellyfish-transaction-builder/__tests__/txn/txn_builder_submit_dfc_htlc.test.ts:20]: You can use testing container from now on

**Review by Jouzo — DISMISSED** [review]
Everything looks great

File: packages/jellyfish-transaction/__tests__/script/dftx/dftx_icxorderbook/ICXSubmitDFCHTLC.test.ts [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+      SmartBuffer.fromBuffer(Buffer.from(hex, 'hex'))
+    )
+
+    const buffer = toBuffer(stack)
+    expect(buffer.toString('hex')).toBe(hex)
+    expect((stack[1] as OP_DEFI_TX).tx.type).toBe(0x33)
+  })
+})
+
+const header = '6a4c514466547833' // OP_RETURN(0x1a) OP_PUSHDATA1(0x4c) (length 81 = 0x51) CDfTx.SIGNATURE(0x44665478) CICXSubmitDFCHTLC.OP_CODE(0x33)
```
Comment (Jouzo) [inline packages/jellyfish-transaction/__tests__/script/dftx/dftx_icxorderbook/ICXSubmitDFCHTLC.test.ts:24]: const header = '6a4c514466547833' // OP_RETURN(0x1a) OP_PUSHDATA1(0x4c) (length 81 = 0x51) CDfTx.SIGNATURE(0x44665478) CICXSubmitDFCHTLC.OP_CODE(0x33)
```suggestion
const header = '6a4c514466547833' // OP_RETURN(0x6a) OP_PUSHDATA1(0x4c) (length 81 = 0x51) CDfTx.SIGNATURE(0x44665478) CICXSubmitDFCHTLC.OP_CODE(0x33)
```

File: packages/jellyfish-transaction-builder/__tests__/txn/txn_builder_submit_dfc_htlc.test.ts (hunk 3 above)
Comment (jingyi2811) [reply packages/jellyfish-transaction-builder/__tests__/txn/txn_builder_submit_dfc_htlc.test.ts:20]: testing.container.start()

File: packages/jellyfish-transaction-builder/__tests__/txn/txn_builder_submit_dfc_htlc.test.ts [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+
+    providers = await getProviders(container)
+    providers.setEllipticPair(WIF.asEllipticPair(GenesisKeys[0].owner.privKey)) // set it to container default
+    builder = new P2WPKHTransactionBuilder(providers.fee, providers.prevout, providers.elliptic)
+
+    // steps required for ICX testing
+    await testing.icxorderbook.setAccounts(await providers.getAddress(), await providers.getAddress())
+    await testing.rpc.account.utxosToAccount({ [testing.icxorderbook.accountDFI]: `${500}@${testing.icxorderbook.symbolDFI}` })
+    await testing.rpc.account.utxosToAccount({ [testing.icxorderbook.accountBTC]: `${10}@${testing.icxorderbook.symbolDFI}` }) // for fee
+    await container.generate(1)
```
Comment (jingyi2811) [inline packages/jellyfish-transaction-builder/__tests__/txn/txn_builder_submit_dfc_htlc.test.ts:31]: testing.generate(1)

File: packages/jellyfish-transaction-builder/__tests__/txn/txn_builder_submit_dfc_htlc.test.ts [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+
+describe('submit DFC HTLC', () => {
+  const container = new MasterNodeRegTestContainer()
+  const testing = Testing.create(container)
+  let providers: MockProviders
+  let builder: P2WPKHTransactionBuilder
+
+  beforeAll(async () => {
+    await container.start()
+    await container.waitForWalletCoinbaseMaturity()
```
Comment (jingyi2811) [inline packages/jellyfish-transaction-builder/__tests__/txn/txn_builder_submit_dfc_htlc.test.ts:21]: testing.container

File: packages/jellyfish-transaction-builder/__tests__/txn/txn_builder_submit_dfc_htlc.test.ts [hunk 7]
```diff
@@ -0,0 +1,177 @@
...
+
+    providers = await getProviders(container)
```
Comment (jingyi2811) [inline packages/jellyfish-transaction-builder/__tests__/txn/txn_builder_submit_dfc_htlc.test.ts:23]: testing.container

Comment (jingyi2811) [reply packages/jellyfish-transaction-builder/__tests__/txn/txn_builder_submit_dfc_htlc.test.ts:23]: And apply to others

**Review by Jouzo — DISMISSED** [review]
All good

**Review by jingyi2811 — APPROVED** [review]

File: packages/jellyfish-transaction/src/script/dftx/index.ts [hunk 8]
```diff
@@ -17,6 +17,7 @@ export * from './dftx_token'
 export * from './dftx_unmapped'
 export * from './dftx_icxorderbook'
 export * from './dftx'
+export * from './dftx_icxorderbook'
```
Comment (surangap) [inline packages/jellyfish-transaction/src/script/dftx/index.ts:20]:
```suggestion
```

**Review by Jouzo — APPROVED** [review]

**Review by fuxingloh — APPROVED** [review]

**Pull request merged**